src/table/table.component.ts

```diff
@@ -302,27 +304,42 @@ const skeletonModel = Table.skeletonModel(0, 0, this.model);
 			const model = new TableModel();
 
 			const header = new Array(columnCount).fill(0).map(() => new TableHeaderItem());
-			const data = new Array(rowCount - 1).fill(0).map(() => new Array(columnCount).fill(0).map(() => new TableItem({ data: " " })));
+			const data = new Array(rowCount).fill(0).map(() => new Array(columnCount).fill(0).map(() => new TableItem({ data: " " })));
 
 			model.header = header;
 			model.data = data;
 			return model;
 		}
 		// If an initial model is provided, to create a skeleton state from the given model,
 		// create only first row with empty data to show loading animation only on the first row.
-		const model = Object.assign( Object.create( Object.getPrototypeOf(initialModel)), initialModel);
-		model.data[0] = model.data[0].map(col => {
-			const tableItem = Object.assign( Object.create( Object.getPrototypeOf(col)), col);
-			tableItem.data = "";
-			return tableItem;
+		const model = new TableModel();
+		model.header = initialModel.header;
+		model.data = initialModel.data;
+
+		model.data[0] = model.data[0].map(column => {
+			column.data = "";
+			return column;
 		});
 
-		for (let i = 1; i < model.data.length; i++) {
-			model.data[i] = model.data[i].map(col => {
-				const tableItem = Object.assign( Object.create( Object.getPrototypeOf(col)), col);
-				tableItem.data = " ";
-				return tableItem;
-			});
+		for (let i = 1; i < (rowCount === -1 ? model.data.length : rowCount); i++) {
+			// Create rows with default `TableItems` for extra rows when given `rowCount` is greater
+			// than the number of rows in the given initial model.
+			if (i >= model.data.length) {
+				model.data[i] = Array(model.data[0].length).fill(0).map(() => new TableItem({ data: " " }));
+			} else {
+				model.data[i] = model.data[i].map(column => {
+					column.data = " ";
+					return column;
+				});
+			}
+		}
+
+		// Delete extra rows if number given `rowCount` is less than the number of rows in a given initial model.
```

Comment:
I'm not sure why we even have row and col count if we're taking in an initial model state? Maybe that should be split out to a separate method?

Reply:
Or add a separate typedef for the method that has it taking the initialModel as the first (only?) argument? and just checking for that